crates/test-programs/wasi-tests/src/bin/path_open_read_without_rights.rs

```diff
@@ -0,0 +1,73 @@
+use std::{env, process};
+use wasi_tests::open_scratch_directory;
+use wasi_tests::{drop_rights, fd_get_rights, create_file};
+
+const TEST_FILENAME: &'static str = "file";
+
+unsafe fn try_read_file(dir_fd: wasi::Fd) {
+    let fd = wasi::path_open(dir_fd, 0, TEST_FILENAME, 0, 0, 0, 0).expect("opening the file");
+
+    // Check that we don't have the right to exeucute fd_read
+    let (rbase, rinher) = fd_get_rights(fd);
+    assert_eq!(
+        rbase & wasi::RIGHTS_FD_READ,
+        0,
+        "should not have base RIGHTS_FD_READ"
+    );
+    assert_eq!(
+        rinher & wasi::RIGHTS_FD_READ,
+        0,
+        "should not have inheriting RIGHTS_FD_READ"
+    );
+
+    let contents = &mut [0u8; 1];
+    let iovec = wasi::Iovec {
+        buf: contents.as_mut_ptr() as *mut _,
+        buf_len: contents.len(),
+    };
+    // Since we no longer have the right to fd_read, trying to read a file
+    // should be an error.
+    wasi::fd_read(fd, &[iovec]).expect_err("reading bytes from file should fail");
```

Comment:
Could you also test for the returned error value here as well?

Reply:
You mean that I should assert that the error code is precisely `ENOTCAPABLE`?

Reply:
Done.